tests/test_utils.py

```diff
@@ -117,6 +121,45 @@ def test_getExpPositionOffset(self):
             self.assertAlmostEqual(result.deltaDec.asDegrees(), deltaDec, 6)
 
 
+class ImageBasedTestCase(lsst.utils.tests.TestCase):
+    """A test case for testing sky position offsets for exposures."""
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            cls.butler = makeDefaultLatissButler()
+        except FileNotFoundError:
+            raise unittest.SkipTest("Skipping tests that require the LATISS butler repo.")
+
+        cls.dataId = {'day_obs': 20200315, 'seq_num': 120, 'detector': 0}
+        cls.bestEffort = BestEffortIsr()  # should always succeed if makeDefaultLatissButler works
+        cls.exp = cls.bestEffort.getExposure(cls.dataId)
+
+    def test_fluxFromFootprint(self):
+        footPrintSet = detectObjectsInExp(self.exp)
+        allFootprints = footPrintSet.getFootprints()
+        self.assertGreaterEqual(len(allFootprints), 3)
+        singleFootprint = allFootprints[0]
+        twoFootprints = allFootprints[0:2]
+
+        # check it can accept a footprintSet, and single and iterables of
+        # footprints
+        for obj in (footPrintSet, singleFootprint, twoFootprints):
+            fluxes = fluxesFromFootprints(obj, self.exp)
+            if isinstance(obj, FootprintSet):
+                expectedLength = len(footPrintSet.getFootprints())
+                self.assertEqual(len(fluxes), expectedLength)  # always one flux per footprint
+
+            self.assertIsInstance(fluxes, Iterable)
+            self.assertIsInstance(fluxes[0], np.floating)  # always returns floats
+
+        # ensure that subtracting the image median from fluxes leave image
+        # functionally unchanged
+        oldExpArray = copy.deepcopy(self.exp.image.array)
+        fluxes = fluxesFromFootprints(obj, self.exp, subtractImageMedian=True)
+        self.assertTrue(np.allclose(self.exp.image.array, oldExpArray))
```

Comment:
Shouldn't this be identically equal? It shouldn't modify the image at all, I'd hope.

Reply:
Eurgh, it's floaty, and subtracting and adding back on a median in some cases, so I was just being careful. Perhaps I need to just do it differently though, and just subtract the median times the area in pixels from the footprint's flux  and never actually touch the pixels at all. That's probably safer, just a little more complex. Let me do it like that instead then.